idea_town/frontend/static-src/app/views/experiment-page.js

```diff
@@ -171,24 +166,29 @@ export default PageView.extend({
     const width = evt.target.offsetWidth;
     evt.target.style.width = width + 'px';
     evt.target.classList.add('state-change');
-    setTimeout(() => {
+
+    this.updateAddon(true, this.model);
+    app.on('webChannel:addon-install:install-ended', () => {
+      this.model.enabled = !this.model.enabled;
       evt.target.classList.remove('state-change');
-      this.updateAddon(true, this.model);
-    }, 2000);
+    });
   },
 
-  uninstall(callback, model) {
+  uninstall(cb, model) {
     const uninstallButton = document.getElementById('uninstall-button');
     const feedbackButton = document.getElementById('feedback-button');
     const width = uninstallButton.offsetWidth;
     uninstallButton.style.width = width + 'px';
     uninstallButton.classList.add('state-change');
     feedbackButton.style.display = 'none';
-    setTimeout(() => {
+
+    cb(false, model);
+
+    app.on('webChannel:addon-uninstall:uninstall-ended', () => {
+      model.enabled = !model.enabled;
       uninstallButton.classList.remove('state-change');
       feedbackButton.style.display = 'initial';
```

Comment:
Hmm, I don't think we need to mess with button styles & such here. [We have these hooked up with data binding to the `enabled` prop of the experiment, no?](https://github.com/mozilla/idea-town/blob/master/idea_town/frontend/static-src/app/views/experiment-page.js#L36) Also, I'm seeing an issue where the "give feedback" button doesn't actually go away on experiment disable - not sure if that's related to this.